Accomodating measurement sets

mj1e16lsst opened the pull request:
## Initial changes to emerlin2caom2. 

The codebase takes a measurement set as an input, extracts the relevant metadata to an xml file, and attempts to upload it to 
The changes are exclusive to python files contained within the emerlin2caom/emerlin2caom directory. As such, there are still some artefacts from the old repository that include: the Dockerfile, the scripts directory, the testing suite. These are left in for now, to be updated when the development environment is finished. 

## Changes in the codebase

The metadata extraction is handled by casa_reader.py and measurement_set_metadata.py. The former handles metadata extracted from the measurement set itself via casa, whereas the latter handles "file" metadata such as size, hashvalue, and locations/names. 

The code for creation of caom observations is contained within main_app.py. It implements the metadata extraction of the other two modules, creates the python object observation, converts it to an XML file, and has command for upload to the repository via curl (not currently functional). 

## Installation and running

Required packages are:

[cadcutils](https://github.com/opencadc/cadctools/tree/main/cadcutils)
[caom2tools](https://github.com/opencadc/caom2tools/tree/main)
[casatools](https://pypi.org/project/casatools/)
[checksumdir](https://pypi.org/project/checksumdir/)

For attempting upload, [Stephen's docker-compose setup](https://github.com/uksrc/caomdev) should be built and running. 

Installation is a little odd right now as the cadc requirements claim to need python>3.8, whereas casatools claims to need python<3.6. Sharon Goliath tested the cadc code with python 3.6 for me and it passed all tests. So install cadcutils and caom2tools in python 3.6, via pip, utilising the --ignore-requires-python option. 

There is a simple test included in "test.py", the xml should be produced but the upload is expected to return a 403 permission denied with the current repository setup.

ebrassfieldbourke commented on emerlin2caom2/casa_reader.py:
```diff
@@ -0,0 +1,46 @@
+
+import casatools
+
+msmd = casatools.msmetadata()
```
In main_app.py, msmd is .measurement_set_metadata, whereas here msmd is the casa msmetadata; potential confusion for package readability.

ebrassfieldbourke commented on emerlin2caom2/main_app.py:
```diff
@@ -71,59 +71,111 @@
+    ReleaseType, ObservationWriter, ProductType, ChecksumURI
 
+import casa_reader as casa
+import measurement_set_metadata as msmd
```
In casa_reader.py, msmd is casatools msmetadata, whereas here msmd is the locally created measurement_set_metadata.py; potential confusion for package readability.

ebrassfieldbourke commented on emerlin2caom2/measurement_set_metadata.py:
Possible re-name eventually as this is mostly for determining filetype and and artefact metadata?

ebrassfieldbourke reviewed (approved):
A few naming confusions for me but otherwise great to merge so i can branch and add more casa functions to read from the measurement sets.

mj1e16lsst commented:
I have added the text from this PR to the README. Although I removed the parts that are more specific to the pull request. 
I added more detailed instructions on installation of the required packages.
I also expanded upon the section where I talk about what the Docker is used for and explicitly stated what it is not used for. 

Without Stephen's docker setup, only the xml creation portion of the code can be expected to run. Perhaps the issue of a build service should be kept in the background until the entirety of the repository can be run and tested.

pahjbo commented:
I agree that we can hold off on the build service for a while, so lets just merge
